### DISCUSSION TITLE
Stuck in "Trying Login with Cached Credentials"

### DISCUSSION THREAD

**Discussion by LOMASKLEW** [SimDynamX/SpaceTeamsPRO#46] [2024 Lunar SAR Competition Q&A] [answered]
Hello, I've been trying to run start the app for a while now, but it keeps getting stuck on this loading screen. I've deleted and redownloaded everything at least 5 times now. Thank you for any help.
![image](https://github.com/user-attachments/assets/7bc8550b-c9cb-439e-8818-f32fbbc6ce40)

**Comment by connorjak** [comment]
There may be a firewall issue or something blocking space teams from getting to our servers for login? We haven't seen this before; usually the login works within 10 seconds. The manual way to reset your cached credentials is by deleting properties.txt in AppData/Local/SpaceCRAFT/Saved

**Comment by rcghpge** [comment]
Hey there, I also am having login issues as well. I moseyed over to the AppData folder but do not see any properties.txt file

**Comment by connorjak** [comment]
Hmm that is odd. I'll take a look at the code. Can both of you send your SpaceTeamsPRO account username?

**Reply by rcghpge** [reply]
rcstputapge

**Reply by rcghpge** [reply]
I tried registering another account to match the emails between the Space Teams Pro app and the website. Thought maybe it would be authentication issues. Still no luck.

**Reply by rcghpge** [reply]
my 2nd account is rcutastppge

**Comment by rcghpge** [comment]
Hello, I am still having issues with installation. A cybersecurity peer of mine suggested it may be a hashing issue? Will keep tinkering away

**Comment by rcghpge** [comment]
I managed to get my desktop environment setup and I can run demos. I am in the initial stages of tinkering with the beta version of the software. However, I noticed that if I move the app/program to another file path like moving it to the desktop, it breaks the program. Will see if I can setup my laptop environment at another time.

**Reply by connorjak** [reply]
Can you tell me what changed to allow you to get past the login? We have one case of reproducing this issue internally but we need all the context we can get.

**Reply by rcghpge** [reply]
The hardware and essentially the software in a way? My laptop still has the caching/login issue, my desktop does not.

**Reply by rcghpge** [reply]
My laptop environment I am still having login issues. At the moment

**Comment by connorjak** [comment]
I've opened an Issue for this bug here: https://github.com/SimDynamX/SpaceTeamsPro/issues/50

**Comment by connorjak** [comment]
Possible fix from one of our testers:

"I logged out of spaceteamspro on the browser so that when I opened the app I had to log in and that seems to have worked"

So ensuring you've logged out on SpaceTeamsPRO.com will help. Also, unsure if this is related for now, but that tester also hadn't yet accepted the trial license on the website.

**Reply by rcghpge** [reply]
I accepted the trial license.

**Comment by connorjak** [comment]
Details of how 0.27.4 affects this: 
- https://github.com/SimDynamX/SpaceTeamsPro/issues/50#issuecomment-2458393794

**Comment by rcghpge** [comment]
The 0.27.3 version allowed me past the frozen caching login issue. I just downloaded the 0.27.4 version and it works! so far 😂 gg

**Comment by connorjak** [comment]
Ok great! 

I'll wait for a second confirmation before marking this issue resolved.

**Reply by connorjak** [reply]
@LOMASKLEW are you able to log in now?

**Comment by LOMASKLEW** [comment]
I will try today, I’m sorry for all the inconvenience.

Thank you very much!

**Comment by LOMASKLEW** [comment]
[image: image.png]Attached is thye screen that I am now frozen on. I am
using a Dell Inspiron 7700 AIO with 11th gen intel core (i7-1165G7). I have
1 terabyte of storage and 32 GB of ram (31.6 usable). I am running windows
11.

**Comment by LOMASKLEW** [comment]
![unnamed](https://github.com/user-attachments/assets/0fdbc0ec-1727-4cea-a79e-27edeb3ea9f2)
Here is the same image in case it isn't visible.

**Reply by connorjak** [reply]
Does your browser open in the background? You may need to alt-tab to get to it

**Comment by LOMASKLEW** [comment]
Yes, we successfully sign in. Then after we click verify and it gets stuck
there.

**Reply by connorjak** [reply]
You will need to also click Validate Login. Is STPRO frozen meaning you can't click on it?

**Reply by connorjak** [reply]
Oh actually I see the Validating... text

**Reply by connorjak** [reply]
I'm unsure why this is happening; @wovw got any ideas on the database side?

**Comment by LOMASKLEW** [comment]
Did you guys ever find a solution? My group and I are still very much
interested in even starting the competition although we are definitely very
behind.

**Reply by connorjak** [reply]
Still looking into this issue. The other fixes seem to have worked for other people; we have one person internally who got a similar issue with 0.27.3 but still need to reproduce this in 0.27.4/0.27.5. Also looking into it on the database side.

**Comment by LOMASKLEW** [comment]
Ah okay thank you very much! We appreciate the opportunity.

**Comment by connorjak** [comment]
Still looking into this; we haven't found the source of the problem yet (app-side or database-side) despite testing with some varied cases. If you can try logging in to STPRO 0.27.5 on another computer, let us know if it works. Both it working and not working on a second computer may tell us something about the problem. Also, if you're able to try logging in from multiple different networks (school network, home network, etc), that would be valuable test data as well.

**Comment by connorjak** [comment, marked as answer]
We believe we've fixed this on our database! Of all things it turned out to be a profanity filter that was rejecting requests (normally should only apply to account creation and renaming). That is the reason why this was specific to certain Space Teams PRO accounts.

**Reply by rcghpge** [reply]
But I don't cuss much 😂 wow of all things. Good work SimDynamX

**Reply by connorjak** [reply]
I think how it affected things was usernames that contain words within 1 or 2 characters of matching profanity. Lots of opportunity for false positives.
